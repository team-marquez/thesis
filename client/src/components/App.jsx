import React from 'react'

import Account from './Account.jsx'
import AllDays from './AllDays.jsx'
import UserProfile from './UserProfile.jsx'
import Onboarding from './Onboarding.jsx'
import UserPreferences from './UserPreferences.jsx'
import firebase from './firebase.js'
import { images } from './helpers/splashImages.js'

import { Sidebar, Menu, Breadcrumb } from 'semantic-ui-react'

class App extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      pickedTrip: false,
      home: false,
      user: 'Welcome User',
      image: 'https://react.semantic-ui.com/images/avatar/large/patrick.png',
      openOnboarding: false,
      visible: false,
      login: false,
      location: 'New York',
      background: '',
      backgroundNY: '',
      trip: 'current'
    }

    this.pickTrip = this.pickTrip.bind(this)
    this.goHome = this.goHome.bind(this)
    this.changeUser = this.changeUser.bind(this)
    this.openOnboarding = this.openOnboarding.bind(this)
    this.closeFirstOnboard = this.closeFirstOnboard.bind(this)

    this.handleButtonClick = this.handleButtonClick.bind(this)
    this.handleSidebarHide = this.handleSidebarHide.bind(this)
    this.handleLogin = this.handleLogin.bind(this)
    this.handleLogout = this.handleLogout.bind(this)

    this.locationChange = this.locationChange.bind(this)
    this.changeToCurrent = this.changeToCurrent.bind(this)
    this.changeToPast = this.changeToPast.bind(this)
    this.goCurrentAndHome = this.goCurrentAndHome.bind(this)
  }

  // Set random background image from our helper splashImage file.
  componentDidMount() {
    let img = images[Math.floor(Math.random() * images.length)]
    console.log(img)
    this.setState({ background: img, backgroundNY: img })
  }

  // Set random background image from our helper splashImage file.
  componentDidMount() {
    let img = images[Math.floor(Math.random() * images.length)]
    console.log(img)
    this.setState({ background: img, backgroundNY: img })
  }

  // Conditional rendering function for displaying the home page and the trip page. Works on LocationModals 'Next' button.
  pickTrip() {
    this.setState({
      pickedTrip: !this.state.pickedTrip
    })
  }

  goHome() {
    this.setState({
      home: false,
      trip: 'current'
    })
  }

  goCurrentAndHome () {
    this.setState({
      home: true,
      trip: 'current',
      pickedTrip: false
    })
  }

  changeToCurrent() {
    this.setState({
      home: true,
      trip: 'current'
    })
  }

  changeToPast() {
    this.setState({
      home: true,
      trip: 'past'
    })
  }

  //Set username
  changeUser(username, image) {
    this.setState({
      user: username,
      image: image || this.state.image
    })
  }

  openOnboarding() {
    this.setState({ openOnboarding: true })
  }

  closeFirstOnboard() {
    this.setState({ openOnboarding: false })
  }

  handleButtonClick() {
    this.setState({ visible: !this.state.visible })
  }

  handleSidebarHide() {
    this.setState({ visible: false })
  }

  handleLogin() {
    this.setState({ login: true })
  }

  handleLogout() {
    this.setState({ login: false })
  }

  locationChange(e) {
    var location = e.target.text

    this.setState({ location })

    if (location === 'New York') {
      this.setState({
        background: this.state.backgroundNY
      })
    } else if (location === 'Paris') {
      this.setState({
        background:
          'https://images.unsplash.com/photo-1500313830540-7b6650a74fd0?ixlib=rb-0.3.5&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max&ixid=eyJhcHBfaWQiOjF9&s=6e938f7571b1c14add60901d6b841307'
      })
    } else if (location === 'Tokyo') {
      this.setState({
        background:
          'https://images.unsplash.com/photo-1493976040374-85c8e12f0c0e?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=ebd34ddde3f2b4ea6dcdc9b7d329b774&auto=format&fit=crop&w=2850&q=80'
      })
    }
  }

  logOut() {
    firebase.auth().signOut()
      .then(() => {
        this.changeUser('Anonymous User', 'https://react.semantic-ui.com/images/avatar/large/patrick.png')
        this.handleLogout()
      })
  }

  render() {
    return (
      <div>

        
        <Sidebar.Pushable>
          <Sidebar
            as={Menu}
            animation="overlay"
            onHide={this.handleSidebarHide}
            vertical
            visible={this.state.visible}
            width="thin"
          >
            <Menu.Item as="a" onClick={this.goHome}>
              Home
            </Menu.Item>
            <Menu.Item as="a" onClick={this.changeToCurrent}>Current Trip</Menu.Item>
            <Menu.Item as='a' onClick={this.changeToPast}>Past Trips</Menu.Item>
            
            {this.state.home === false &&             
            <Menu.Item as="a" onClick={this.logOut}>
              Logout
            </Menu.Item> }
          </Sidebar>

          <Sidebar.Pusher>
            {this.state.home === false ? (
              <div>
                {this.state.pickedTrip === false ? (
                  <div className='splash'>
                    <img className='splashImage' src={this.state.background}/>
                    <div>
                      <Account
                        user={this.state.user}
                        trip={this.state.trip}
                        styled={{height: '9%', width: '5%', position: 'absolute', top: '3%', left: '93%', cursor:'pointer'}}
                        image={this.state.image}
                        changeUser={this.changeUser}
                        openOnboarding={this.openOnboarding}
                        handleButtonClick={this.handleButtonClick}
                        handleLogin={this.handleLogin}
                        handleLogout={this.handleLogout}
                        loggedIn={this.state.login}
                      />
                    </div>
                    <div className='cityName'>
                      {this.state.location === 'New York' ? (
                        <div>
                          <div className='newYork'>New York</div>
                          <div className='newYorkTrip'>
                            <UserPreferences pickTrip={this.pickTrip} able={false}/>
                          </div>
                        </div>
                      ) : this.state.location === 'Tokyo' ? (
                        <div>
                          <div className='tokyo'>Tokyo</div>
                          <div className='tokyoTrip'>
                            <UserPreferences pickTrip={this.pickTrip} able={true}/>
                          </div>
                        </div>
                      ) : (<div>
                        <div className='paris'>Paris</div>
                        <div className='parisTrip'>
                          <UserPreferences pickTrip={this.pickTrip} able={true}/>
                        </div>
                        </div>)}
                    </div>
                    <Onboarding open={this.state.openOnboarding} closer = {this.closeFirstOnboard} />
                  </div>
                ) : (
                  <div className='allDaysComp'>
                    <AllDays home={this.goHome} user={this.state.user} goCurrentAndHome={this.goCurrentAndHome}/>
                  </div>
                )}
              </div>
            ) : (
              <div>
                <UserProfile home={this.goHome} user={this.state.user} image={this.state.image} trip={this.state.trip}/>
                <Account
                  styled={{height: '13%', width: '4%', position: 'absolute', top: '1%', left: '94%', cursor:'pointer'}}
                  styled2={{height: '3%', width: '4%', position: 'absolute', top: '5px', left: '94%', cursor:'pointer'}}
                  trip={this.state.trip}
                  user={this.state.user}
                  image={this.state.image}
                  changeUser={this.changeUser}
                  openOnboarding={this.openOnboarding}
                  handleButtonClick={this.handleButtonClick}
                  handleLogin={this.handleLogin}
                  handleLogout={this.handleLogout}
                  loggedIn={this.state.login}
                />
              </div>
            )}

            {(this.state.pickedTrip === false && this.state.home === false) ? (
              <div className='breadCrumbs'>
                <Breadcrumb>
                  <Breadcrumb.Section onClick={this.locationChange}>
                    Tokyo
                  </Breadcrumb.Section>
                  <Breadcrumb.Divider icon="map pin" />
                  <Breadcrumb.Section onClick={this.locationChange}>
                    Paris
                  </Breadcrumb.Section>
                  <Breadcrumb.Divider icon="map pin" />
                  <Breadcrumb.Section onClick={this.locationChange}>
                    New York
                  </Breadcrumb.Section>
                  <Breadcrumb.Divider icon="map pin" />
                </Breadcrumb>
              </div>
            ) : (<div></div>)}
          </Sidebar.Pusher>
        </Sidebar.Pushable>
      </div>
    )
  }
}

export default App
