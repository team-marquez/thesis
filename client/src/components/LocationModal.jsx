import React from "react";
import moment from 'moment';
import { Slider } from "react-semantic-ui-range";
import {
  Grid,
  Modal,
  Input,
  Label,
  Checkbox,
  Button,
  Image
} from "semantic-ui-react";
import { format, addDays } from "date-fns";
import { DateRange } from "react-date-range";
import gql from "graphql-tag";
import { Query } from "react-apollo";
import { ApolloConsumer } from 'react-apollo';


let PREF_QUERY = gql`
  query PREF_QUERY($pref: Pref){
    userPrefs(pref: $pref)
  }
`;

Date.prototype.addDays = function(days) {
  var date = new Date(this.valueOf());
  date.setDate(date.getDate() + days);
  return date;
}

class LocationModal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      bright: "brightness(100%)",
      f_a: 50,
      i_o: 50,
      l_t: 50,
      NSFW: false,
      party_size: 1,
      budget: 0,
      tripDates: null,
      maxDate: null,
    };
    this.mouseEnter = this.mouseEnter.bind(this);
    this.mouseExit = this.mouseExit.bind(this);
  }

  handleSelect(ranges){
		console.log(ranges);
	}

  mouseEnter() {
    this.setState({
      bright: "brightness(125%)"
    });
  }

  mouseExit() {
    this.setState({
      bright: "brightness(100%)"
    });
  }

  render() {

    const selectionRange = {
			startDate: new Date(),
			endDate: new Date(),
			key: 'selection',
		}
    return (
      <ApolloConsumer>
      {(client) => { return (
      <div className="locationOptions" style={{ width: "100%" }}>
        <Grid>
          <Grid.Row centered columns={2}>
            <Grid.Column>
              <Modal
                size={"mini"}
                trigger={
                  <Image
                    src="https://drscdn.500px.org/photo/119659039/m%3D900/v2?webp=true&sig=f0973f50f2fc640b2c6a13491efb40365c11c425b81b6dc05f7dbe9bf15c0d87"
                    onMouseEnter={this.mouseEnter}
                    onMouseLeave={this.mouseExit}
                    style={{
                      marginLeft: "10px",
                      marginBottom: "10px",
                      filter: this.state.bright
                    }}
                  />
                }
              >
                <Modal.Header style={{ textAlign: "center" }}>
                  Set Your Preferences
                </Modal.Header>
                <Modal.Content>
                  <div style={{ textAlign: "center" }}>
                    <DateRange
                      maxDate={moment(this.state.maxDate)}
                      onChange={(changes) =>{
                      this.setState({
                        maxDate: changes.startDate._d.addDays(3),
                        tripDates: changes
                      })}}
                      calendars={1}
                    />
                  </div>
                  <br/>
                  <div style={{ textAlign: "center" }}>
                    <Input
                      labelPosition="right"
                      type="number"
                      placeholder="Budget"
                      onChange={(e) => this.setState({
                        budget: e.target.value
                      }) }
                    >
                      <Label basic>$</Label>
                      <input />
                      <Label>.00</Label>
                    </Input>
                    <br />
                  </div>
                  <br/>


                  <Modal.Description>
                    <div style={{ float: "right", display: "inline-block" }}>
                      {"Activities" + ` ${this.state.f_a}%`}
                    </div>
                    <div style={{ textAlign: "left", display: "inline-block" }}>
                      {"Food" + ` ${100 - this.state.f_a}%`}
                    </div>
                    <Slider
                      value={this.state.f_a}
                      color="purple"
                      inverted={false}
                      settings={{
                        start: 50,
                        min: 0,
                        max: 100,
                        step: 1,
                        onChange: value => {
                          this.setState({
                            f_a: value
                          });
                        }
                      }}
                    />
                    <div style={{ float: "right", display: "inline-block" }}>
                      {"Outdoors" + ` ${this.state.i_o}%`}
                    </div>
                    <div style={{ textAlign: "left", display: "inline-block" }}>
                      {"Indoors" + ` ${100 - this.state.i_o}%`}
                    </div>
                    <Slider
                      value={this.state.i_o}
                      color="pink"
                      inverted={false}
                      settings={{
                        start: 50,
                        min: 0,
                        max: 100,
                        step: 1,
                        onChange: value => {
                          this.setState({
                            i_o: value
                          });
                        }
                      }}
                    />
                    <div style={{ float: "right", display: "inline-block" }}>
                      {"Touristy" + ` ${this.state.l_t}%`}
                    </div>
                    <div style={{ textAlign: "left", display: "inline-block" }}>
                      {"Local" + ` ${100 - this.state.l_t}%`}
                    </div>
                    <Slider
                      value={this.state.l_t}
                      color="green"
                      inverted={false}
                      settings={{
                        start: 50,
                        min: 0,
                        max: 100,
                        step: 1,
                        onChange: value => {
                          this.setState({
                            l_t: value
                          });
                        }
                      }}
                    />{" "}
                    <br />
                    <div
                      style={{ display: "inline-block", marginRight: "95px" }}
                    >
                      <Input
                        icon="users"
                        size="mini"
                        iconPosition="left"
                        type="number"
                        placeholder="Party size"
                        style={{ width: "100px", textAlign: "right" }}
                        onChange={(e) => this.setState({
                          party_size: e.target.value
                        })}
                      />
                    </div>
                    <div style={{ display: "inline-block" }}>
                      <Checkbox label="Family Friendly" onChange={() => this.setState({NSFW: !this.state.NSFW})} />
                    </div>
                    <div style={{ textAlign: "right" }}>
                      <br />
                      <Button
                        content="Next"
                        icon="right arrow"
                        labelPosition="right"
                        onClick={async () => {
                          console.log('client', this.state.tripDates.endDate)
                          const { data } = await client.query({
                            query: PREF_QUERY,
                            variables: { pref: {
                              totalBudget: this.state.budget,
                              partySize: this.state.party_size,
                              tripDates: {
                                startDate: this.state.tripDates.startDate,
                                endDate: this.state.tripDates.endDate._d.addDays(-1)
                              },
                              LT: this.state.l_t,
                              IO: this.state.i_o,
                              FA: this.state.f_a,
                              kidFriendly: this.state.NSFW
                            }}
                          })
                          client.writeData({data})          
                          this.props.pickTrip()
                        }}
                      />
                    </div>
                  </Modal.Description>
                </Modal.Content>
              </Modal>
            </Grid.Column>
            <Grid.Column>
              <Image
                src="https://drscdn.500px.org/photo/92890847/m%3D900/v2?webp=true&sig=51138635394c2a7a8a38178eb09e2fb88cd0bebcbcd90ad84f1d3265b8b00e22"
                style={{
                  marginBottom: "10px",
                  marginLeft: "-10px",
                  opacity: "0.8",
                  filter: "grayscale(100%)"
                }}
              />
            </Grid.Column>
            <Grid.Column>
              <Image
                src="https://drscdn.500px.org/photo/3236342/m%3D900/v2?webp=true&sig=9c44f7758ff90da55d211411839b469f3d2e4652ade16897dbe76fd3a911411a"
                style={{
                  marginLeft: "10px",
                  opacity: "0.8",
                  filter: "grayscale(100%)"
                }}
              />
            </Grid.Column>
            <Grid.Column>
              <Image
                src="https://drscdn.500px.org/photo/52784332/m=900/9831e3f3de4586bd113ab838ce2aaeb7"
                style={{
                  marginLeft: "-10px",
                  opacity: "0.8",
                  filter: "grayscale(100%)"
                }}
              />
            </Grid.Column>
          </Grid.Row>
        </Grid>
      </div>)}}
      </ApolloConsumer>
    );
  }
}

export default LocationModal;
