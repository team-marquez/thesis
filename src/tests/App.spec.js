import React from 'react';
import {shallow} from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import App from '../../client/src/components/App.jsx';
import LoginButton from "../../client/src/components/LoginButton.jsx";
import LocationModal from "../../client/src/components/LocationModal.jsx";
import Onboarding from '../../client/src/components/Onboarding.jsx'

configure({ adapter: new Adapter() });

describe('App', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should render a <div />', () => {
    expect(wrapper.find('div').length).toEqual(5);
  });
  it('should render the LoginButton Component', () => {
    expect(wrapper.containsMatchingElement(<LoginButton />)).toEqual(true);
  });
  it('should render the LocationModal Component', () => {
    expect(wrapper.containsMatchingElement(<LocationModal />)).toEqual(true);
  });
  it('should render the Onboarding Component', () => {
    expect(wrapper.containsMatchingElement(<Onboarding />)).toEqual(true);
  });
});