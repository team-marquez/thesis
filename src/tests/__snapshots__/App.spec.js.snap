// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <div>
                <div
                        style={
                                Object {
                                        "marginBottom": "15px",
                                        "textAlign": "center",
                                      }
                        }
                >
                        <LoginButton
                                changeUser={[Function]}
                                home={[Function]}
                                image="https://react.semantic-ui.com/images/avatar/large/patrick.png"
                                openOnboarding={[Function]}
                                user="Welcome User"
                        />
                </div>
                <div
                        style={
                                Object {
                                        "display": "inline-block",
                                      }
                        }
                >
                        <LocationModal
                                pickTrip={[Function]}
                        />
                </div>
                <Footer />
                <Onboarding
                        closer={[Function]}
                        open={false}
                />
        </div>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <div
                    style={
                              Object {
                                        "marginBottom": "15px",
                                        "textAlign": "center",
                                      }
                    }
          >
                    <LoginButton
                              changeUser={[Function]}
                              home={[Function]}
                              image="https://react.semantic-ui.com/images/avatar/large/patrick.png"
                              openOnboarding={[Function]}
                              user="Welcome User"
                    />
          </div>
          <div
                    style={
                              Object {
                                        "display": "inline-block",
                                      }
                    }
          >
                    <LocationModal
                              pickTrip={[Function]}
                    />
          </div>
          <Footer />
          <Onboarding
                    closer={[Function]}
                    open={false}
          />
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              style={
                            Object {
                                          "marginBottom": "15px",
                                          "textAlign": "center",
                                        }
              }
>
              <LoginButton
                            changeUser={[Function]}
                            home={[Function]}
                            image="https://react.semantic-ui.com/images/avatar/large/patrick.png"
                            openOnboarding={[Function]}
                            user="Welcome User"
              />
</div>,
            <div
              style={
                            Object {
                                          "display": "inline-block",
                                        }
              }
>
              <LocationModal
                            pickTrip={[Function]}
              />
</div>,
            <Footer />,
            <Onboarding
              closer={[Function]}
              open={false}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <LoginButton
                changeUser={[Function]}
                home={[Function]}
                image="https://react.semantic-ui.com/images/avatar/large/patrick.png"
                openOnboarding={[Function]}
                user="Welcome User"
/>,
              "style": Object {
                "marginBottom": "15px",
                "textAlign": "center",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "changeUser": [Function],
                "home": [Function],
                "image": "https://react.semantic-ui.com/images/avatar/large/patrick.png",
                "openOnboarding": [Function],
                "user": "Welcome User",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <LocationModal
                pickTrip={[Function]}
/>,
              "style": Object {
                "display": "inline-block",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "pickTrip": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "closer": [Function],
              "open": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <div>
                    <div
                              style={
                                        Object {
                                                  "marginBottom": "15px",
                                                  "textAlign": "center",
                                                }
                              }
                    >
                              <LoginButton
                                        changeUser={[Function]}
                                        home={[Function]}
                                        image="https://react.semantic-ui.com/images/avatar/large/patrick.png"
                                        openOnboarding={[Function]}
                                        user="Welcome User"
                              />
                    </div>
                    <div
                              style={
                                        Object {
                                                  "display": "inline-block",
                                                }
                              }
                    >
                              <LocationModal
                                        pickTrip={[Function]}
                              />
                    </div>
                    <Footer />
                    <Onboarding
                              closer={[Function]}
                              open={false}
                    />
          </div>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            <div
                        style={
                                    Object {
                                                "marginBottom": "15px",
                                                "textAlign": "center",
                                              }
                        }
            >
                        <LoginButton
                                    changeUser={[Function]}
                                    home={[Function]}
                                    image="https://react.semantic-ui.com/images/avatar/large/patrick.png"
                                    openOnboarding={[Function]}
                                    user="Welcome User"
                        />
            </div>
            <div
                        style={
                                    Object {
                                                "display": "inline-block",
                                              }
                        }
            >
                        <LocationModal
                                    pickTrip={[Function]}
                        />
            </div>
            <Footer />
            <Onboarding
                        closer={[Function]}
                        open={false}
            />
</div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                style={
                                Object {
                                                "marginBottom": "15px",
                                                "textAlign": "center",
                                              }
                }
>
                <LoginButton
                                changeUser={[Function]}
                                home={[Function]}
                                image="https://react.semantic-ui.com/images/avatar/large/patrick.png"
                                openOnboarding={[Function]}
                                user="Welcome User"
                />
</div>,
              <div
                style={
                                Object {
                                                "display": "inline-block",
                                              }
                }
>
                <LocationModal
                                pickTrip={[Function]}
                />
</div>,
              <Footer />,
              <Onboarding
                closer={[Function]}
                open={false}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <LoginButton
                  changeUser={[Function]}
                  home={[Function]}
                  image="https://react.semantic-ui.com/images/avatar/large/patrick.png"
                  openOnboarding={[Function]}
                  user="Welcome User"
/>,
                "style": Object {
                  "marginBottom": "15px",
                  "textAlign": "center",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "changeUser": [Function],
                  "home": [Function],
                  "image": "https://react.semantic-ui.com/images/avatar/large/patrick.png",
                  "openOnboarding": [Function],
                  "user": "Welcome User",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <LocationModal
                  pickTrip={[Function]}
/>,
                "style": Object {
                  "display": "inline-block",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "pickTrip": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "closer": [Function],
                "open": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
